inmoose/sim/splat.py

```diff
@@ -0,0 +1,195 @@
+    to https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1305-0.
+
+    Arguments
+    ---------
+    nb_genes : int
+        number of genes
+    nb_samples : int
+        number of samples
+    batch : array-like, optional
+        batch indices. Must have as many elements as `nb_samples`.
+    group : array-like, optional
+        vector/factor for biological condition of interest. Must have as many
+        elements as `nb_samples`.
+    dropout : bool, optional
+        if True, simulate scRNASeq data by adding drop-outs. Defaults to `False`.
+
+    alpha : float, optional
+        shape of the gamma distribution to draw the initial means from
+    beta : float, optional
+        rate of the gamme distribution to draw the initial means from
+    outlier_prob : float, optional
+        proportion of outliers
+    outlier_location : float, optional
+        location parameter for the log-normal distribution to draw the outlier
+        inflation factors from
+    outlier_scale : float, optional
+        scale parameter for the log-normal distribution to draw the outlier
+        inflation factors from
+    libsize_loc : float, optional
+        location of the log-normal distribution to draw the library sizes from
+    libsize_scale : float, optional
+        scale of the log-normal distribution to draw the library sizes from
+    phi : float, optional
+        common dispersion
+    bcv_df : int, optional
+        degrees of freedom in the inverse chi-square distribution used to draw
+        the biological coefficient of variation parameters
+    x0 : float, optional
+        dropout midpoint, used to draw the dropouts. Unused if
+        `single_cell=False`.
+    k : float, optional
+        dropout shape, used to draw the dropouts. Unused if `single_cell=False`.
+    random_state : {int, `numpy.random.Generator` or `numpy.random.RandomState`},
+                    optional
+        If `random_state` is None (or `np.random`), the
+        `numpy.random.RandomState` singleton is used.  If `random_state` is an
+        int, a new ``RandomState`` instance is used, seeded with
+        `random_state`.  If `random_state` is already a ``Generator`` or
+        ``RandomState`` instance, that instance is used.
+
+    Returns
+    -------
+    ndarray
+        simulated count matrix, of size `nb_genes` x `nb_samples`
+    """
+
+    # make sure we have a Generator
+    random_state = np.random.default_rng(random_state)
+
+    if batch is None:
+        batch = np.zeros(nb_samples, dtype=np.int64)
+    batch = np.asarray(batch)
+    assert len(batch) == nb_samples
+
+    if group is None:
+        group = np.zeros(nb_samples, dtype=np.int64)
+    group = np.asarray(group)
+    assert len(group) == nb_samples
+
+    nb_batches = len(np.unique(batch))
+    nb_groups = len(np.unique(group))
+
+    # draw gene means
+    original_means = gamma.rvs(alpha, beta, size=nb_genes, random_state=random_state)
```

Comment:
Not sure here but is beta really the rate of the distribution if set like that?
Looks like rvs would need a 1/beta value for its scale parameter which is the third parameter, what do you think?

Reply:
Nice catch! `rgamma(N, shape=shape, rate=rate)` in R is indeed equivalent to `gamma.rvs(shape, scale=1/rate, size=N)` in Python.